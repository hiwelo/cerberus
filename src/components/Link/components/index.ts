export { default as LinkMarkup } from './LinkMarkup';
