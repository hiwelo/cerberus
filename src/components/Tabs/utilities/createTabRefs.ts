import { useRef } from 'react';
import { TabsProps } from '../types';

/**
 * Creates a series of React Ref for each panel part of the Tabs defined for
 * this `Tabs` component
 *
 * @param tabs The dictionary of Tabs for the related Tabs component
 */
export function createTabRefs(tabs: TabsProps['tabs']): TabsProps['tabs'] {
  // Transforms the TabListDictionary as an array of tabs
  const tabsEntries = Object.entries(tabs);
  const tabsWithRefs: TabsProps['tabs'] = {};

  // Loops over each tab and creates a React Ref for each panel HTMLElement
  tabsEntries.forEach(([tabKey, tabInfo]) => {
    const refPanel = useRef<HTMLDivElement>(null);

    tabsWithRefs[tabKey] = { ...tabInfo, refPanel };
  });

  return tabsWithRefs;
}
