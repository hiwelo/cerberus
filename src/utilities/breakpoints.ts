/** List of breakpoints in use in our styled components */
export const breakpoints = {
  xs: '0px',
  md: '540px',
};
