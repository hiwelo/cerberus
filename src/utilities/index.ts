export { breakpoints } from './breakpoints';
export { listNodesFromConnection } from './listNodesFromConnection';
