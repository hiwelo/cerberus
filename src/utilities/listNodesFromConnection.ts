/** Describes the generic structure of a GraphQL edge */
type GenericEdge<T> = {
  /** Item from a GraphQL connection object */
  node: T;
  /** Potential typing for the edge added by GraphQL */
  __typename: string;
};

/** Describes the generic structure of a GraphQL connection object */
type GenericConnection<T> = {
  edges: GenericEdge<T>[];
};

/**
 * Returns an array of all nodes contained in the provided GraphQL connection
 * object
 *
 * @param connection GraphQL connection object containing the nodes to return
 */
export function listNodesFromConnection<T>(
  connection?: GenericConnection<T>,
): T[] | undefined {
  return connection?.edges.map(({ node }) => node);
}
