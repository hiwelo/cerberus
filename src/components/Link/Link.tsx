import React, { FunctionComponent } from 'react';
import { LinkMarkup } from './components';
import { LinkProps } from './types';

export const Link: FunctionComponent<LinkProps> = ({
  children,
  target,
  ...rest
}) => {
  return (
    <LinkMarkup
      rel={target === '_blank' ? 'noopener noreferrer' : undefined}
      target={target}
      {...rest}
    >
      {children}
    </LinkMarkup>
  );
};
