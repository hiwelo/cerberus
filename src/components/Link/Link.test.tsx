import React from 'react';
import { testA11y, testSnapshot } from 'utilities/test';
import { Link } from './Link';

const TestContainer = <Link href="#">Test</Link>;

describe('Link', () => {
  testSnapshot(TestContainer);
  testSnapshot(
    <Link href="#" target="_blank">
      Test with target blank
    </Link>,
    'with target blank',
  );
  testA11y(TestContainer);
});
