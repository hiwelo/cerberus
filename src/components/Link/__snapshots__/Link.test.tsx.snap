// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should match snapshot 1`] = `
.c0 {
  position: relative;
  color: #006e97;
  font-weight: 500;
}

.c0:focus {
  outline: none;
}

<a
  class="c0"
  href="#"
>
  Test
</a>
`;

exports[`Link should match snapshot: with target blank 1`] = `
.c0 {
  position: relative;
  color: #006e97;
  font-weight: 500;
}

.c0:focus {
  outline: none;
}

<a
  class="c0"
  href="#"
  rel="noopener noreferrer"
  target="_blank"
>
  Test with target blank
</a>
`;
